internal/service/appsync/resolver.go

```diff
@@ -34,40 +34,58 @@ func ResourceResolver() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
-			"type": {
+			"arn": {
 				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Computed: true,
 			},
-			"field": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+			"caching_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
```

Comment:
👋 been going through PRs to learn and I've noticed a lot of these where the type is TypeList and the MaxItems is 1 item of type map. We then have expand functions which always expects a list of one map item. What is this pattern? Why can't the type be TypeMap instead?